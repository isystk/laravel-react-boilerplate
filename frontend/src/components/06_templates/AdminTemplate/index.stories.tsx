import { Meta, Story } from '@storybook/react'
import React from 'react'
import AdminTemplate, { AdminTemplateProps } from './index'
import MainService from '@/services/main'

export default {
  title: '06_templates/AdminTemplate',
  component: AdminTemplate,
} as Meta

const Template: Story = () => {
  const main = new MainService(() => ({}))
  const props: AdminTemplateProps = { main, title: 'サンプル' }
  return <AdminTemplate {...props} />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
