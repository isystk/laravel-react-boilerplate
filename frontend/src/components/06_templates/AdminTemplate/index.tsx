import Circles from '@/components/02_interactions/Circles'
import React, { FC, useState } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { connect } from '@/components/hoc'
import HtmlSkeleton, {
  NoIndex,
  Title,
} from '@/components/05_layouts/HtmlSkeleton'
import MainService from '@/services/main'
import { adminMenuItems } from '@/constants/menu'
import Logo from '@/components/01_atoms/Logo'
import { Url } from '@/constants/url'
import { useI18n } from '@/components/i18n'
import { useRouter } from 'next/router'
import Breadcrumb, {
  type BreadcrumbItem,
} from '@/components/01_atoms/Breadcrumb'
import DropDown from '@/components/01_atoms/DropDown'
import Hamburger from '@/components/01_atoms/Hamburger'
import SideMenu from '@/components/04_organisms/SideMenu'

/** AdminTemplateProps Props */
export type AdminTemplateProps = WithChildren & {
  main: MainService
  title: string
  breadcrumb?: BreadcrumbItem[]
}
/** Presenter Props */
export type PresenterProps = AdminTemplateProps & {
  t
  logout
  router
  isMenuOpen
  setMenuOpen
}

/** Presenter Component */
const AdminTemplatePresenter: FC<PresenterProps> = ({
  children,
  main,
  t,
  title,
  breadcrumb = [],
  logout,
  router,
  isMenuOpen,
  setMenuOpen,
  ...props
}) => (
  <HtmlSkeleton>
    <Title>{`${title}－Admin`}</Title>
    <NoIndex />
    <div className="h-screen">
      <div className="h-16 bg-base p-4 md:p-0 flex">
        <div className="flex pr-3 z-50 md:hidden">
          <Hamburger isMenuOpen={isMenuOpen} setMenuOpen={setMenuOpen} />
        </div>
        <Logo link={Url.AdminHome} />
        {main.user && (
          <div className="ml-auto p-3">
            <DropDown
              label={main.user.userName}
              items={[{ label: 'ログアウト', link: () => logout() }]}
            />
          </div>
        )}
      </div>
      <div className="flex">
        <SideMenu
          isMenuOpen={isMenuOpen}
          setMenuOpen={setMenuOpen}
          menuItems={adminMenuItems}
        />
        <div className="w-full">
          <div
            className="flex items-center justify-center"
            style={{ height: 'calc(100vh - 4rem)' }}
          >
            <Circles>
              {main.user && <Breadcrumb items={breadcrumb} />}
              <div className="py-8 md:p-8 m-auto">{children}</div>
            </Circles>
          </div>
        </div>
      </div>
    </div>
  </HtmlSkeleton>
)

/** Container Component */
const AdminTemplateContainer: React.FC<
  ContainerProps<AdminTemplateProps, PresenterProps>
> = ({ presenter, main, children, ...props }) => {
  const router = useRouter()
  const { t } = useI18n('Common')
  const [isMenuOpen, setMenuOpen] = useState(false)

  const logout = async () => {
    await main.logout()
    location.reload()
  }

  return presenter({
    children,
    main,
    t,
    logout,
    router,
    isMenuOpen,
    setMenuOpen,
    ...props,
  })
}

export default connect<AdminTemplateProps, PresenterProps>(
  'AdminTemplate',
  AdminTemplatePresenter,
  AdminTemplateContainer
)
