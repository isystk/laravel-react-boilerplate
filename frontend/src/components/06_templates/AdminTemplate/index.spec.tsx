import React from 'react'
import { render } from '@testing-library/react'
import AdminTemplate, { AdminTemplateProps } from './index'
import MainService from '@/services/main'

describe('LandingPageTemplate', () => {
  it('Match Snapshot', () => {
    const main = new MainService(() => ({}))
    const props: AdminTemplateProps = { main, title: 'サンプル' }
    const { asFragment } = render(<AdminTemplate {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
