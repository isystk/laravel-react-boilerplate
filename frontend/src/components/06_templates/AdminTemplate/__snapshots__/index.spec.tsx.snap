// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageTemplate Match Snapshot 1`] = `
<DocumentFragment>
  <div
    class="h-screen"
  >
    <div
      class="h-16 bg-base p-4 md:p-0 flex"
    >
      <div
        class="flex pr-3 z-50 md:hidden"
      >
        <button
          class="space-y-2 css-jgxxba"
        >
          <div
            class="w-6 h-0.5 bg-black"
          />
          <div
            class="w-6 h-0.5 bg-black"
          />
          <div
            class="w-6 h-0.5 bg-black"
          />
        </button>
      </div>
      <a
        class="css-0"
        href="#"
      >
        <div
          class="w-20 md:w-40"
        >
          <img
            class="css-s08p0c "
            src="/images/logo.png"
          />
        </div>
      </a>
    </div>
    <div
      class="flex"
    >
      <div
        class="css-12c2knp p-4 border-t z-50 h-screen bg-base css-fmjhrv"
      >
        <ul
          class="list-none"
        >
          <li
            class="my-6"
          >
            <a
              class="break-words whitespace-pre-wrap text-gray-700 font-bold whitespace-nowrap"
              href="http://localhost:9000/admin/product"
              rel=""
            >
              Product List
            </a>
          </li>
          <li
            class="my-6"
          >
            <a
              class="break-words whitespace-pre-wrap text-gray-700 font-bold whitespace-nowrap"
              href="http://localhost:9000/admin/subscriber"
              rel=""
            >
              Subscribe List
            </a>
          </li>
        </ul>
      </div>
      <div
        class="hidden z-20 opacity-50 fixed top-0 w-full h-screen"
      />
      <div
        class="w-full"
      >
        <div
          class="flex items-center justify-center"
          style="height: calc(100vh - 4rem);"
        >
          <div
            class="css-1wwfi4"
          >
            <ul
              class="css-10ffb9g"
            >
              <li
                class="css-i6ammj"
              />
              <li
                class="css-l07tar"
              />
              <li
                class="css-1hre2z7"
              />
              <li
                class="css-p2xyi8"
              />
              <li
                class="css-u8edpl"
              />
              <li
                class="css-ik3prs"
              />
              <li
                class="css-1wh69e5"
              />
              <li
                class="css-1cg3at6"
              />
              <li
                class="css-1iaru34"
              />
              <li
                class="css-1io9drp"
              />
            </ul>
            <div
              class="relative"
            >
              <div
                class="py-8 md:p-8 m-auto"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
